In [1]:
import pandas as pd
import matplotlib.pyplot as plt
df = pd.read_csv('/content/Dataset .csv')


In [2]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9551 entries, 0 to 9550
Data columns (total 21 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Restaurant ID         9551 non-null   int64  
 1   Restaurant Name       9551 non-null   object 
 2   Country Code          9551 non-null   int64  
 3   City                  9551 non-null   object 
 4   Address               9551 non-null   object 
 5   Locality              9551 non-null   object 
 6   Locality Verbose      9551 non-null   object 
 7   Longitude             9551 non-null   float64
 8   Latitude              9551 non-null   float64
 9   Cuisines              9542 non-null   object 
 10  Average Cost for two  9551 non-null   int64  
 11  Currency              9551 non-null   object 
 12  Has Table booking     9551 non-null   object 
 13  Has Online delivery   9551 non-null   object 
 14  Is delivering now     9551 non-null   object 
 15  Switch to order menu 

In [ ]:
Online_delivery_counts = df['Has Online delivery'].value_counts()
total_Restaurants = len(df)
Online_delivery_percentage = Online_delivery_counts / total_Restaurants * 100
print(Online_delivery_percentage)


No     74.337766
Yes    25.662234
Name: Has Online delivery, dtype: float64


In [ ]:
online_delivery_percentage_yes = Online_delivery_percentage.loc['Yes']
print(online_delivery_percentage_yes)


25.662234321013504


In [ ]:
print(df.columns)

Index(['Restaurant ID', 'Restaurant Name', 'Country Code', 'City', 'Address',
       'Locality', 'Locality Verbose', 'Longitude', 'Latitude', 'Cuisines',
       'Average Cost for two', 'Currency', 'Has Table booking',
       'Has Online delivery', 'Is delivering now', 'Switch to order menu',
       'Price range', 'Aggregate rating', 'Rating color', 'Rating text',
       'Votes'],
      dtype='object')


In [ ]:
online_delivery_ratings = df.groupby('Has Online delivery')['Aggregate rating'].mean()
print(online_delivery_ratings)


Has Online delivery
No     2.465296
Yes    3.248837
Name: Aggregate rating, dtype: float64


# Based on our results :

--> The percentage of restaurants that offer online delivery : 25.662234321013504 = 25.6%

-->The average rating of restaurants
   with online delivery = 3.248837
   without online delivery = 2.465296